auto-self-nominate/index.ts

```diff
@@ -0,0 +1,179 @@
+    const pubkey = buffer.toString("hex");
+    return PlatformAddress.fromPublic(pubkey, {
+        networkId: sdk.networkId,
+    });
+}
+
+function decodeNumber(buffer: Buffer): number {
+    return parseInt("0x" + buffer.toString("hex"), 16);
+}
+
+async function getCurrentTermId(bestBlockNumber: number): Promise<number | null> {
+    return new Promise((resolve, reject) => {
+        sdk.rpc
+            .sendRpcRequest("chain_getTermMetadata", [bestBlockNumber])
+            .then(result => {
+                if (result === null) {
+                    resolve(null);
+                }
+                const [prevTermLastBlockNumber, currentTermId] = result;
+                if (
+                    typeof prevTermLastBlockNumber === "number" &&
+                    typeof currentTermId === "number"
+                ) {
+                    resolve(currentTermId);
+                }
+                reject(
+                    Error(
+                        `Expected getTermMetadata to return [number, number] | null but it returned ${result}`,
+                    ),
+                );
+            })
+            .catch(reject);
+    });
+}
+
+async function sendSelfNominateTransaction(
+    accountInfo: { account: string; passphrase: string },
+    params: { deposit: U64; metadata: string },
+) {
+    const tx = createSelfNomination(params);
+    const { account, passphrase } = accountInfo;
+    await sdk.rpc.account.sendTransaction({
+        tx,
+        account,
+        passphrase,
+    });
+}
+
+function createSelfNomination(params: { deposit: U64; metadata: string }): Custom {
+    const { deposit, metadata } = params;
+    const handlerId = new U64(STAKE_ACTION_HANDLER_ID);
+    const actionTag = ACTION_TAG_SELF_NOMINATE;
+    const bytes = RLP.encode([actionTag, deposit, metadata]);
+    return new Custom(
+        {
+            handlerId,
+            bytes,
+        },
+        networkId(SERVER),
+    );
+}
+
+async function needsNomination(
+    address: string,
+    nominationFrequencyLevel: number,
+): Promise<boolean> {
+    const bestBlockNumber = await sdk.rpc.chain.getBestBlockNumber()!;
+    const candidates: Map<string, number> = await (async () => {
+        const retval = new Map();
+        const data = (await sdk.rpc.engine.getCustomActionData(
+            2,
+            ["Candidates"],
+            bestBlockNumber,
+        ))!;
+        const list: Buffer[][] = RLP.decode(Buffer.from(data, "hex"));
+        list.map(([encodedPubkey, _, encodedNominationEnd]) =>
+            retval.set(
+                decodePlatformAddressFromPubkey(encodedPubkey).value,
+                decodeNumber(encodedNominationEnd),
+            ),
+        );
+        return retval;
+    })();
+    const nominationEndAt = candidates.get(address);
+    if (nominationEndAt == null) {
+        throw new Error(`SelfNominate first with specific deposit vakue before repeating`);
+    } else {
```

Comment:
You don't need to use `else`. The if block always throws an error.

Reply:
Fixed